app/src/main/java/io/appium/uiautomator2/server/mjpeg/MjpegScreenshotServer.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.uiautomator2.server.mjpeg;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import io.appium.uiautomator2.utils.Logger;
+
+public class MjpegScreenshotServer extends Thread {
+    private final List<MjpegScreenshotClient> clients = new ArrayList<>();
+    private final MjpegScreenshotStream mjpegScreenshotStream =
+        new MjpegScreenshotStream(clients);
+    private boolean isStopped = false;
+    private int port;
```

Comment:
final?

Reply:
Updated references to things that will not be re-assigned across all the Mjpeg* classes.

This will be resolved w/ the upcoming update(s)